action.yml

```
@@ -205,6 +250,39 @@ runs:
       path: .tmp-ansible-collection-checkout
       persist-credentials: false
       ref: ${{ inputs.git-checkout-ref }}
+      fetch-depth: >-
+        ${{
+          steps.compute-change-detection-coverage.outputs.change-detection-arg
```

This should probably be
```suggestion
          fromJSON(steps.compute-change-detection-coverage.outputs.change-detection-arg)
```
because GHA treats inputs as strings, and it's necessary to turn them into real boolean objects for this to behave properly.

---

The output is either an empty string, or it is `--changed --base-branch main` or something similar. Other places in the file also use the same syntax to check whether something is empty or not.

---

In regular unconditional render uses — yes, but in conditional clauses I use this conversion because it prevents implicit unexpected behaviors.

---

I don't see how a random string can be parsed as JSON. `--changed --base-branch main` is not JSON. If it would have quotes it would be, but it does not have quotes.

---

Okay, I usually use this for values that are supposed to contain boolean literals.